devices/HumanDataCollector/HumanDataCollector.cpp

```diff
@@ -331,12 +566,93 @@ void HumanDataCollector::run()
+        pImpl->jointTorquesVec = pImpl->iHumanDynamics->getJointTorques();
+
+    }
+
+    // Prepare buffer vectors
+    // Handle arrays to vectors
+    if (pImpl->isAttached.stateProvider) {
+
+        pImpl->basePoseArray[0] = pImpl->basePosition[0];
+        pImpl->basePoseArray[1] = pImpl->basePosition[1];
+        pImpl->basePoseArray[2] = pImpl->basePosition[2];
+        pImpl->basePoseArray[3] = pImpl->baseOrientation[0];
+        pImpl->basePoseArray[4] = pImpl->baseOrientation[1];
+        pImpl->basePoseArray[5] = pImpl->baseOrientation[2];
+        pImpl->basePoseArray[6] = pImpl->baseOrientation[3];
+
+        pImpl->basePoseVec = std::vector<double>(pImpl->basePoseArray.begin(), pImpl->basePoseArray.end());
+
+        pImpl->baseVeclocityVec = std::vector<double>(pImpl->baseVelocity.begin(), pImpl->baseVelocity.end());
+
+        pImpl->comPositionVec = std::vector<double>(pImpl->comPosition.begin(), pImpl->comPosition.end());
+
+        pImpl->comVelocityVec = std::vector<double>(pImpl->comVelocity.begin(), pImpl->comVelocity.end());
+
+        pImpl->comProperAccInBaseFrameVec = std::vector<double>(pImpl->comProperAccInBaseFrame.begin(), pImpl->comProperAccInBaseFrame.end());
+
+        pImpl->comProperAccInWorldFrameVec = std::vector<double>(pImpl->comProperAccInWorldFrame.begin(), pImpl->comProperAccInWorldFrame.end());
+
+    }
+
+    // Update interface data to matio buffers
+    if (pImpl->matioLogger) {
+
+        if (pImpl->isAttached.stateProvider) {
+
+            // Set state joint names once
+            if (pImpl->humanDataStruct.stateJointNames.empty()) {
+                pImpl->humanDataStruct.stateJointNames = pImpl->stateJointNames;
+            }
+
+
+            pImpl->humanDataStruct.data.at("basePose").push_back(pImpl->basePoseVec);
+            pImpl->humanDataStruct.data.at("baseVelocity").push_back(pImpl->baseVeclocityVec);
+
+            pImpl->humanDataStruct.data.at("comPosition").push_back(pImpl->comPositionVec);
+            pImpl->humanDataStruct.data.at("comVelocity").push_back(pImpl->comVelocityVec);
+            pImpl->humanDataStruct.data.at("comProperAccelerationInBaseFrame").push_back(pImpl->comProperAccInBaseFrameVec);
+            pImpl->humanDataStruct.data.at("comProperAccelerationInWorldFrame").push_back(pImpl->comProperAccInWorldFrameVec);
+
+            pImpl->humanDataStruct.data.at("jointPositions").push_back(pImpl->jointPositionsVec);
+            pImpl->humanDataStruct.data.at("jointVelocities").push_back(pImpl->jointVelocitiesVec);
+
+
+        }
+
+        if (pImpl->isAttached.wrenchProvider) {
+
+            // Set wrench measurements source names once
+            if (pImpl->humanDataStruct.wrenchMeasurementSourceNames.empty()) {
+                pImpl->humanDataStruct.wrenchMeasurementSourceNames = pImpl->wrenchMeasurementSourceNames;
+            }
+
+            pImpl->humanDataStruct.data["wrenchMeasurements"].push_back(pImpl->wrenchMeasurementValuesVec);
```

Comment:
Sometimes you use `at()` and other you use the `[]` operator. Is there any reason?

Reply:
No particular reason, Corrected in https://github.com/robotology/human-dynamics-estimation/pull/181/commits/7613f51dfe14f3376831210feea7290d7447f82e